src/components/modals/SettingsModal.tsx

```diff
@@ -68,6 +72,29 @@ export const SettingsModal = memo(function SettingsModal({
     }
   };
 
+  const handleChangeModel = async (v: string) => {
+    if (await checkModelAccess(v)) {
```

Comment:
should we add a loading state with the `<Spinner /> here? User experience feels less snappy when selecting new model with small delay.